docs/architecture/blazor-for-web-forms-developers/data.md

```diff
@@ -1,22 +1,134 @@
+
+You can then generate the database table appropriate for this class using the following commands:
+
+```dotnetcli
+dotnet ef migrations add 'Create Product table'
+dotnet ef database update
+```
+
+The first command defines the changes you're making to the database schema as a new EF Migration called `Create Product table`.  A Migration defines how to apply and remove your new database changes.
+
+Once applied, you have a simple `Product` table in your database and some new classes added to the project that help manage the database schema.  You can find these generated classes, by default, in a new folder called *Migrations*.  When you make changes to the `Product` class or add more related classes you would like interacting with your database, you need to run the command-line commands again with a new name of the migration.  This command will generate another set of migration classes to update your database schema.
+
+### EF Database First
+
+For existing databases, you can generate the classes for EF Core by using the .NET command-line tools. To scaffold the classes, use a variation of the following command:
+
+```dotnetcli
+dotnet ef dbcontext scaffold "CONNECTION STRING" Microsoft.EntityFrameworkCore.SqlServer -c MyDbContext -t Product -t Customer
+```
+
+The preceding command connects to the database using the specified connection string and the `Microsoft.EntityFrameworkCore.SqlServer` provider. Once connected, a database context class named `MyDbContext` is created. Additionally, supporting classes are created for the `Product` and `Customer` tables that were specified with the `-t` options. There are many configuration options for this command to generate the class hierarchy appropriate for your database. For a complete reference, see [the command's documentation](/ef/core/miscellaneous/cli/dotnet#dotnet-ef-dbcontext-scaffold).
+
+More information about [EF Core](/ef/core/) can be found on the Microsoft Docs site.
+
+## Interact with web services
+
+When ASP.NET was first released, SOAP services were the preferred way for web servers and clients to exchange data. Much has changed since that time, and the preferred interactions with services have shifted to direct HTTP client interactions. With ASP.NET Core and Blazor, you can register the configuration of your `HttpClient` in the `Startup` class's `ConfigureServices` method. Use that configuration when you need to interact with the HTTP endpoint. Consider the following configuration code:
+
+```csharp
+services.AddHttpClient("github", c =>
+{
+    c.BaseAddress = new Uri("http://api.github.com/");
+    // Github API versioning
+    c.DefaultRequestHeaders.Add("Accept", "application/vnd.github.v3+json");
+    // Github requires a user-agent
+    c.DefaultRequestHeaders.Add("User-Agent", "BlazorWebForms-Sample");
+});
```

Comment:
```suggestion
services.AddHttpClient("github", client =>
{
    client.BaseAddress = new Uri("http://api.github.com/");
    // Github API versioning
    client.DefaultRequestHeaders.Add("Accept", "application/vnd.github.v3+json");
    // Github requires a user-agent
    client.DefaultRequestHeaders.Add("User-Agent", "BlazorWebForms-Sample");
});
```

I believe it would be easier for readers, even experienced C# devs - for the `c` variable to be named `client` instead. As a reader I wouldn't have to look at the `.AddHttpClient` function name again to try to infer what the lambda parameter type is, instead the name is self-documenting.